### PR TITLE
feat: add a newline after commands that don't end their output with one

### PR DESCRIPTION
**Pull request by chankruze**
Issue: #776

### CODE REVIEW

**Comment by antongolub** [comment]
@chankruze,

1. There should be a test.
2. No need to convert the whole buffer to string to check the last symbol only.
```js
const eol = Buffer.from('\n', 'utf-8')

if (!entry.data.slice(-1).equals(eol)) {
  entry.data = Buffer.concat([entry.data, eol])
}
```

**Comment by chankruze** [comment]
@antongolub What's your opinion about this ?

> This method is not compatible with the Uint8Array.prototype.slice(), which is a superclass of Buffer. To copy the slice, useUint8Array.prototype.slice().

```js
const eol = Buffer.from('\n', 'utf-8');
if (Uint8Array.prototype.slice.call(entry.data, -1)[0] !== eol[0]) {
  entry.data = Buffer.concat([entry.data, eol]);
}
```

**Comment by antongolub** [comment]
Well, it seems too verbose to modify the value. Let's keep the initial approach. Just move this helper to util.ts and add a test plz.
```ts
export function ensureEol (buffer: Buffer): Buffer {
  if (buffer.toString('utf8', buffer.length - 1) !== '\n') {
    return Buffer.concat([buffer, Buffer.from('\n')])
  }

  return buffer
}
```

**Comment by chankruze** [comment]
@antongolub sure! will do that.

**Comment by chankruze** [comment]
@antongolub what about this approach ?

```js
export function ensureEol(buffer: Buffer): Buffer {
  if (!buffer.toString('utf8').endsWith('\n')) {
    return Buffer.concat([buffer, Buffer.from('\n')])
  }

  return buffer
}
```

**Comment by chankruze** [comment]
@antongolub please review the changes and comment if anything needs to be done.

File: test/util.test.js [hunk 1]
```diff
@@ -165,3 +166,20 @@ test('tempfile() creates temporary files', () => {
   assert.match(tf, /\/zx-.+\/bar\.txt$/)
   assert.equal(fs.readFileSync(tf, 'utf-8'), 'bar')
 })
+
+test('should ensure buffer ends with a newline character', () => {
```
Comment (antongolub) [inline test/util.test.js:170]: `test('ensureEol() ...')`

**Review by antongolub — COMMENTED** [review]
mention the method

File: test/util.test.js (hunk 1 above)
Comment (chankruze) [reply test/util.test.js]: @antongolub fixed!

**Pull request merged**